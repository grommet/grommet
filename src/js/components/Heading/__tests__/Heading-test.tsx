import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';
import { hpe } from 'grommet-theme-hpe';

import { Grommet } from '../../Grommet';
import { Heading } from '..';

test('Heading renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Heading renders outside grommet wrapper', () => {
  const { container } = render(<Heading />);

  expect(container.firstChild).toMatchSnapshot();
});

test('Heading accepts ref', () => {
  const ref = React.createRef<HTMLHeadingElement>();
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading ref={ref} />
    </Grommet>,
  );

  expect(ref.current).not.toBeNull();
  expect(container.firstChild).toMatchSnapshot();
});

test('Heading level renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading level={1} />
      <Heading level={2} />
      <Heading level={3} />
      <Heading level={4} />
      <Heading level="1" />
      <Heading level="2" />
      <Heading level="3" />
      <Heading level="4" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Heading size renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading level={1} size="small" />
      <Heading level={1} size="medium" />
      <Heading level={1} size="large" />
      <Heading level={1} size="xlarge" />
      <Heading level={2} size="small" />
      <Heading level={2} size="medium" />
      <Heading level={2} size="large" />
      <Heading level={2} size="xlarge" />
      <Heading level={3} size="small" />
      <Heading level={3} size="medium" />
      <Heading level={3} size="large" />
      <Heading level={3} size="xlarge" />
      <Heading level={4} size="small" />
      <Heading level={4} size="medium" />
      <Heading level={4} size="large" />
      <Heading level={4} size="xlarge" />
      <Heading level={1} size="77px" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Heading textAlign renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading textAlign="start" />
      <Heading textAlign="center" />
      <Heading textAlign="end" />
      <Heading textAlign="justify" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Heading margin renders', () => {
  const { container } = render(
    <Grommet>
      <Heading margin="small" />
      <Heading margin="medium" />
      <Heading margin="large" />
      <Heading margin="none" />
      <Heading margin={{ bottom: 'small' }} />
      <Heading margin={{ top: 'small' }} />
      <Heading margin={{ bottom: 'none' }} />
      <Heading margin={{ top: 'none' }} />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Heading color renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading color="brand" />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

const LONG = 'a b c d e f g h i j k l m n o p q r s t u v w x y z';

test('Heading truncate renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading truncate={false}>{LONG}</Heading>
      <Heading truncate>{LONG}</Heading>
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('responsive renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading responsive />
      <Heading responsive={false} />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('weight renders', () => {
  const { asFragment } = render(
    <Grommet theme={hpe}>
      <Heading>My heading</Heading>
      <Heading weight="normal">My heading</Heading>
      <Heading weight="bold">My heading</Heading>
      <Heading weight={700}>My heading</Heading>
    </Grommet>,
  );

  expect(asFragment()).toMatchSnapshot();
});

test('Theme based font family renders', () => {
  const customTheme = {
    heading: {
      font: {
        family: 'Fira Sans',
      },
      level: {
        1: {
          font: {
            family: 'Arial',
          },
        },
        2: {
          font: {
            family: 'Roboto',
          },
        },
        3: {
          font: {
            family: 'Ubuntu',
          },
        },
      },
    },
  };
  const { container } = render(
    <Grommet theme={customTheme}>
      <Heading level={1} />
      <Heading level={2} />
      <Heading level={3} />
      <Heading level={4} />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Theme based font weight renders', () => {
  const customTheme = {
    heading: {
      weight: 600,
      level: {
        1: {
          font: {
            weight: '700',
          },
        },
        2: {
          font: {
            weight: '400',
          },
        },
        3: {
          font: {
            weight: '200',
          },
        },
      },
    },
  };
  const { container } = render(
    <Grommet theme={customTheme}>
      <Heading level={1} />
      <Heading level={2} />
      <Heading level={3} />
      <Heading level={4} />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Theme color renders', () => {
  const customTheme = {
    heading: {
      color: 'text-strong',
    },
  };
  const { container } = render(
    <Grommet theme={customTheme}>
      <Heading level={1} />
      <Heading level={2} />
      <Heading level={3} />
      <Heading level={4} />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});

test('Throws a warning when heading.level is undefined in the theme.', () => {
  global.console.warn = jest.fn();

  const customTheme = {
    heading: {
      level: {
        6: undefined,
      },
    },
  };

  render(
    <Grommet theme={customTheme}>
      <Heading level={6} />
    </Grommet>,
  );

  const consoleMsg = 'Heading level 6 is not defined in your theme.';
  expect(global.console.warn).toHaveBeenCalledWith(consoleMsg);
});

test('Heading fill renders', () => {
  const { container } = render(
    <Grommet theme={hpe}>
      <Heading fill />
    </Grommet>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
