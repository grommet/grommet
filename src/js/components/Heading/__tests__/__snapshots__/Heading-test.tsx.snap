// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Heading accepts ref 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
</div>
`;

exports[`Heading color renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #01a982;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
</div>
`;

exports[`Heading fill renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: none;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: none;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
</div>
`;

exports[`Heading level renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c2 {
  font-size: 1.75rem;
  line-height: 2rem;
  max-width: 816px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c3 {
  font-size: 1.5rem;
  line-height: 1.75rem;
  max-width: 624px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c4 {
  font-size: 1.25rem;
  line-height: 1.5rem;
  max-width: 432px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 1.5rem;
    line-height: 1.75rem;
    max-width: 624px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 1.25rem;
    line-height: 1.5rem;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 1.125rem;
    line-height: 1.25rem;
    max-width: 384px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
  <h2
    class="c2"
  />
  <h3
    class="c3"
  />
  <h4
    class="c4"
  />
  <h1
    class="c1"
  />
  <h2
    class="c2"
  />
  <h3
    class="c3"
  />
  <h4
    class="c4"
  />
</div>
`;

exports[`Heading margin renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  margin: 12px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c2 {
  margin: 24px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c3 {
  margin: 48px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c4 {
  margin: 0px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c5 {
  margin-bottom: 12px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c6 {
  margin-top: 12px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c7 {
  margin-bottom: 0px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c8 {
  margin-top: 0px;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

@media only screen and (max-width: 768px) {
  .c1 {
    margin: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    margin: 12px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    margin: 24px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    margin: 0px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c5 {
    margin-bottom: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c5 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c6 {
    margin-top: 6px;
  }
}

@media only screen and (max-width: 768px) {
  .c6 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c7 {
    margin-bottom: 0px;
  }
}

@media only screen and (max-width: 768px) {
  .c7 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c8 {
    margin-top: 0px;
  }
}

@media only screen and (max-width: 768px) {
  .c8 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
  <h1
    class="c2"
  />
  <h1
    class="c3"
  />
  <h1
    class="c4"
  />
  <h1
    class="c5"
  />
  <h1
    class="c6"
  />
  <h1
    class="c7"
  />
  <h1
    class="c8"
  />
</div>
`;

exports[`Heading renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
</div>
`;

exports[`Heading renders outside grommet wrapper 1`] = `
.c0 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

@media only screen and (max-width: 768px) {
  .c0 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

<h1
  class="c0"
/>
`;

exports[`Heading size renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 1.75rem;
  line-height: 2rem;
  max-width: 816px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c2 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c3 {
  font-size: 3rem;
  line-height: 3rem;
  max-width: 1968px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c4 {
  font-size: 3.75rem;
  line-height: 3.75rem;
  max-width: 2736px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c5 {
  font-size: 1.5rem;
  line-height: 1.75rem;
  max-width: 624px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c6 {
  font-size: 3rem;
  line-height: 3rem;
  max-width: 1584px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c7 {
  font-size: 1.25rem;
  line-height: 1.5rem;
  max-width: 528px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c8 {
  font-size: 1.5rem;
  line-height: 1.75rem;
  max-width: 624px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c9 {
  font-size: 1.75rem;
  line-height: 2rem;
  max-width: 816px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c10 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1008px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c11 {
  font-size: 1.125rem;
  line-height: 1.25rem;
  max-width: 432px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c12 {
  font-size: 1.25rem;
  line-height: 1.5rem;
  max-width: 432px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c13 {
  font-size: 1.5rem;
  line-height: 1.75rem;
  max-width: 432px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c14 {
  font-size: 1.75rem;
  line-height: 2rem;
  max-width: 432px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c15 {
  font-size: 77px;
  line-height: normal;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.5rem;
    line-height: 1.75rem;
    max-width: 624px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 2.25rem;
    line-height: 2.5rem;
    max-width: 1200px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 3rem;
    line-height: 3rem;
    max-width: 1584px;
  }
}

@media only screen and (max-width: 768px) {
  .c5 {
    font-size: 1.25rem;
    line-height: 1.5rem;
    max-width: 528px;
  }
}

@media only screen and (max-width: 768px) {
  .c6 {
    font-size: 2.25rem;
    line-height: 2.5rem;
    max-width: 1008px;
  }
}

@media only screen and (max-width: 768px) {
  .c7 {
    font-size: 1.125rem;
    line-height: 1.25rem;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c8 {
    font-size: 1.25rem;
    line-height: 1.5rem;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c9 {
    font-size: 1.5rem;
    line-height: 1.75rem;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c10 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c11 {
    font-size: 1rem;
    line-height: 1.125rem;
    max-width: 384px;
  }
}

@media only screen and (max-width: 768px) {
  .c12 {
    font-size: 1.125rem;
    line-height: 1.25rem;
    max-width: 384px;
  }
}

@media only screen and (max-width: 768px) {
  .c13 {
    font-size: 1.25rem;
    line-height: 1.5rem;
    max-width: 384px;
  }
}

@media only screen and (max-width: 768px) {
  .c14 {
    font-size: 1.5rem;
    line-height: 1.75rem;
    max-width: 384px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
  <h1
    class="c2"
  />
  <h1
    class="c3"
  />
  <h1
    class="c4"
  />
  <h2
    class="c5"
  />
  <h2
    class="c1"
  />
  <h2
    class="c2"
  />
  <h2
    class="c6"
  />
  <h3
    class="c7"
  />
  <h3
    class="c8"
  />
  <h3
    class="c9"
  />
  <h3
    class="c10"
  />
  <h4
    class="c11"
  />
  <h4
    class="c12"
  />
  <h4
    class="c13"
  />
  <h4
    class="c14"
  />
  <h1
    class="c15"
  />
</div>
`;

exports[`Heading textAlign renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  text-align: left;
  color: #2e2e2e;
}

.c2 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  text-align: center;
  color: #2e2e2e;
}

.c3 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  text-align: right;
  color: #2e2e2e;
}

.c4 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  text-align: justify;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
  <h1
    class="c2"
  />
  <h1
    class="c3"
  />
  <h1
    class="c4"
  />
</div>
`;

exports[`Heading truncate renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c2 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </h1>
  <h1
    class="c2"
  >
    a b c d e f g h i j k l m n o p q r s t u v w x y z
  </h1>
</div>
`;

exports[`Theme based font family renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-family: Arial;
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c2 {
  font-family: Roboto;
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c3 {
  font-family: Ubuntu;
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  overflow-wrap: break-word;
}

.c4 {
  font-family: Fira Sans;
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
  overflow-wrap: break-word;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
  <h2
    class="c2"
  />
  <h3
    class="c3"
  />
  <h4
    class="c4"
  />
</div>
`;

exports[`Theme based font weight renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 700;
  overflow-wrap: break-word;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 400;
  overflow-wrap: break-word;
}

.c3 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 200;
  overflow-wrap: break-word;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
  overflow-wrap: break-word;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
  <h2
    class="c2"
  />
  <h3
    class="c3"
  />
  <h4
    class="c4"
  />
</div>
`;

exports[`Theme color renders 1`] = `
.c0 {
  font-size: 18px;
  line-height: 24px;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 50px;
  line-height: 56px;
  max-width: 1200px;
  font-weight: 600;
  overflow-wrap: break-word;
  color: #000000;
}

.c2 {
  font-size: 34px;
  line-height: 40px;
  max-width: 816px;
  font-weight: 600;
  overflow-wrap: break-word;
  color: #000000;
}

.c3 {
  font-size: 26px;
  line-height: 32px;
  max-width: 624px;
  font-weight: 600;
  overflow-wrap: break-word;
  color: #000000;
}

.c4 {
  font-size: 18px;
  line-height: 24px;
  max-width: 432px;
  font-weight: 600;
  overflow-wrap: break-word;
  color: #000000;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 34px;
    line-height: 40px;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 26px;
    line-height: 32px;
    max-width: 624px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 18px;
    line-height: 24px;
    max-width: 432px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 16px;
    line-height: 22px;
    max-width: 384px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
  <h2
    class="c2"
  />
  <h3
    class="c3"
  />
  <h4
    class="c4"
  />
</div>
`;

exports[`responsive renders 1`] = `
.c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c2 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

<div
  class="c0"
>
  <h1
    class="c1"
  />
  <h1
    class="c2"
  />
</div>
`;

exports[`weight renders 1`] = `
<DocumentFragment>
  .c0 {
  font-family: 'Metric',Arial,sans-serif;
  font-size: 1.125rem;
  line-height: 1.5rem;
  background-color: #ffffff;
  color: #555555;
  box-sizing: border-box;
  -webkit-text-size-adjust: 100%;
  -ms-text-size-adjust: 100%;
  -moz-osx-font-smoothing: grayscale;
  -webkit-font-smoothing: antialiased;
}

.c1 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 400;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c2 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: normal;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c3 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: bold;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

.c4 {
  font-size: 2.25rem;
  line-height: 2.5rem;
  max-width: 1200px;
  font-weight: 700;
  overflow-wrap: break-word;
  color: #2e2e2e;
}

@media only screen and (max-width: 768px) {
  .c1 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c2 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c3 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

@media only screen and (max-width: 768px) {
  .c4 {
    font-size: 1.75rem;
    line-height: 2rem;
    max-width: 816px;
  }
}

<div
    class="c0"
  >
    <h1
      class="c1"
    >
      My heading
    </h1>
    <h1
      class="c2"
    >
      My heading
    </h1>
    <h1
      class="c3"
    >
      My heading
    </h1>
    <h1
      class="c4"
    >
      My heading
    </h1>
  </div>
</DocumentFragment>
`;
